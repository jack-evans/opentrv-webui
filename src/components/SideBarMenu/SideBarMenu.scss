$sideBarMenu-width: 300px;

.SideBarMenu {
  background-color: $ui-03;
  height: calc(100vh - 60px);
  left: -$sideBarMenu-width;
  position: absolute;
  top: 60px;
  transition: left .3s ease-in-out;
  width: $sideBarMenu-width;
  z-index: 10000;
}

.SideBarMenu__open {
  box-shadow: 0 2px 10px 0;
  left: 0;
}