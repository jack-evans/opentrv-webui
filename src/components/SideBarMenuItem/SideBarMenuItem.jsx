import React, { Component } from 'react'

class SideBarMenuItem extends Component {
  render () {
    return (
      <div>
        SideBarMenuItem
      </div>
    )
  }
}

export default SideBarMenuItem
