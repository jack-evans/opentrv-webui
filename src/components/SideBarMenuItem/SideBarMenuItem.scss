
.SideBarMenuItem {
  align-items: center;
  display: flex;
  height: 50px;
  justify-content: flex-start;
  padding-left: 40px;
  padding-right: 10px;
  width: 250px;

  .SideBarMenuItem__content {
    color: #000;
    text-align: left;
    text-decoration: none;
  }
}