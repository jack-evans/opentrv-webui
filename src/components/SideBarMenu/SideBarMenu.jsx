import React, { Component } from 'react'
import SideBarMenuItem from '../SideBarMenuItem/SideBarMenuItem'
import './SideBarMenu.scss'

// Content to be populated in the side bar menu
const sideBarMenuContent = [
  {
    href: 'https://facebook.github.io/react/',
    text: 'About React'
  },
  {
    href: 'https://httpbin.org',
    text: 'httpbin.org'
  }
]

class SideBarMenu extends Component {
  /**
   * render method
   *
   * renders the side bar menu component of the UI experience
   * @returns {HTML} - SideBarMenu component
   */
  render () {
    const sideBarMenuClassName = this.props.isOpen ? 'SideBarMenu SideBarMenu__open' : 'SideBarMenu'
    const sideBarMenuItems = sideBarMenuContent.map((sideBarMenuItem, index) => {
      return (<SideBarMenuItem key={index} href={sideBarMenuItem.href} text={sideBarMenuItem.text} />)
    })

    return (
      <div className={sideBarMenuClassName}>
        {sideBarMenuItems}
      </div>
    )
  }
}

export default SideBarMenu
