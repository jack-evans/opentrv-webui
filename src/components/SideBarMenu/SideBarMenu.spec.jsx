import React from 'react'
import { shallow } from 'enzyme'
import SideBarMenu from './SideBarMenu'
import SideBarMenuItem from '../SideBarMenuItem/SideBarMenuItem'

describe('SideBarMenu.jsx', () => {
  it('renders without crashing', () => {
    const expectedRender = (
      <div className='SideBarMenu'>
        <SideBarMenuItem key={0} href='https://facebook.github.io/react/' text='About React' />
        <SideBarMenuItem key={1} href='https://httpbin.org' text='httpbin.org' />
      </div>
    )
    expect(shallow(<SideBarMenu isOpen={false} />).contains(expectedRender)).toBe(true)
  })

  describe('when isOpen is true', () => {
    it('renders with the SideBarMenu__open class', () => {
      const expectedRender = (
        <div className='SideBarMenu SideBarMenu__open'>
          <SideBarMenuItem key={0} href='https://facebook.github.io/react/' text='About React' />
          <SideBarMenuItem key={1} href='https://httpbin.org' text='httpbin.org' />
        </div>
      )
      expect(shallow(<SideBarMenu isOpen />).contains(expectedRender)).toBe(true)
    })
  })
})
