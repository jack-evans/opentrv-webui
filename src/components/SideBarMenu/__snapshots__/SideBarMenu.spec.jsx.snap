// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarMenu.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBarMenu
    isOpen={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SideBarMenuItem
          href="http://opentrv.org.uk/what-is-opentrv/"
          text="About OpenTRV"
        />,
        <SideBarMenuItem
          href="http://opentrv.org.uk/principles/"
          text="OpenTRV principles"
        />,
        <SideBarMenuItem
          href="https://twitter.com/OpenTRV"
          text="OpenTRV Twitter"
        />,
        <SideBarMenuItem
          href="http://opentrv.org.uk/documents/"
          text="TRV documentation"
        />,
        <SideBarMenuItem
          href="https://github.com/jack-evans/opentrv-webui/issues"
          text="Report an Issue"
        />,
      ],
      "className": "SideBarMenu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "href": "http://opentrv.org.uk/what-is-opentrv/",
          "text": "About OpenTRV",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "href": "http://opentrv.org.uk/principles/",
          "text": "OpenTRV principles",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "href": "https://twitter.com/OpenTRV",
          "text": "OpenTRV Twitter",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "href": "http://opentrv.org.uk/documents/",
          "text": "TRV documentation",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "href": "https://github.com/jack-evans/opentrv-webui/issues",
          "text": "Report an Issue",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideBarMenuItem
            href="http://opentrv.org.uk/what-is-opentrv/"
            text="About OpenTRV"
          />,
          <SideBarMenuItem
            href="http://opentrv.org.uk/principles/"
            text="OpenTRV principles"
          />,
          <SideBarMenuItem
            href="https://twitter.com/OpenTRV"
            text="OpenTRV Twitter"
          />,
          <SideBarMenuItem
            href="http://opentrv.org.uk/documents/"
            text="TRV documentation"
          />,
          <SideBarMenuItem
            href="https://github.com/jack-evans/opentrv-webui/issues"
            text="Report an Issue"
          />,
        ],
        "className": "SideBarMenu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "href": "http://opentrv.org.uk/what-is-opentrv/",
            "text": "About OpenTRV",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "href": "http://opentrv.org.uk/principles/",
            "text": "OpenTRV principles",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "href": "https://twitter.com/OpenTRV",
            "text": "OpenTRV Twitter",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "href": "http://opentrv.org.uk/documents/",
            "text": "TRV documentation",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "href": "https://github.com/jack-evans/opentrv-webui/issues",
            "text": "Report an Issue",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SideBarMenu.jsx when isOpen is true renders with the SideBarMenu__open class 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBarMenu
    isOpen={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SideBarMenuItem
          href="http://opentrv.org.uk/what-is-opentrv/"
          text="About OpenTRV"
        />,
        <SideBarMenuItem
          href="http://opentrv.org.uk/principles/"
          text="OpenTRV principles"
        />,
        <SideBarMenuItem
          href="https://twitter.com/OpenTRV"
          text="OpenTRV Twitter"
        />,
        <SideBarMenuItem
          href="http://opentrv.org.uk/documents/"
          text="TRV documentation"
        />,
        <SideBarMenuItem
          href="https://github.com/jack-evans/opentrv-webui/issues"
          text="Report an Issue"
        />,
      ],
      "className": "SideBarMenu SideBarMenu__open",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "href": "http://opentrv.org.uk/what-is-opentrv/",
          "text": "About OpenTRV",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "href": "http://opentrv.org.uk/principles/",
          "text": "OpenTRV principles",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "href": "https://twitter.com/OpenTRV",
          "text": "OpenTRV Twitter",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "href": "http://opentrv.org.uk/documents/",
          "text": "TRV documentation",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "href": "https://github.com/jack-evans/opentrv-webui/issues",
          "text": "Report an Issue",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideBarMenuItem
            href="http://opentrv.org.uk/what-is-opentrv/"
            text="About OpenTRV"
          />,
          <SideBarMenuItem
            href="http://opentrv.org.uk/principles/"
            text="OpenTRV principles"
          />,
          <SideBarMenuItem
            href="https://twitter.com/OpenTRV"
            text="OpenTRV Twitter"
          />,
          <SideBarMenuItem
            href="http://opentrv.org.uk/documents/"
            text="TRV documentation"
          />,
          <SideBarMenuItem
            href="https://github.com/jack-evans/opentrv-webui/issues"
            text="Report an Issue"
          />,
        ],
        "className": "SideBarMenu SideBarMenu__open",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "href": "http://opentrv.org.uk/what-is-opentrv/",
            "text": "About OpenTRV",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "href": "http://opentrv.org.uk/principles/",
            "text": "OpenTRV principles",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "href": "https://twitter.com/OpenTRV",
            "text": "OpenTRV Twitter",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "href": "http://opentrv.org.uk/documents/",
            "text": "TRV documentation",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "href": "https://github.com/jack-evans/opentrv-webui/issues",
            "text": "Report an Issue",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
